lib/replicate/checkpointer.js

```diff
@@ -5,24 +5,52 @@ var explain404 = require('./../deps/explain404');
 var pouchCollate = require('pouchdb-collate');
 var collate = pouchCollate.collate;
 
-function updateCheckpoint(db, id, checkpoint, returnValue) {
+var REPLICATION_ID_VERSION = 2;
+
+function updateCheckpoint(db, id, checkpoint, session, returnValue) {
   return db.get(id).catch(function (err) {
     if (err.status === 404) {
       if (db.type() === 'http') {
         explain404('PouchDB is just checking if a remote checkpoint exists.');
       }
-      return {_id: id};
+      return {
+        session_id: session,
+        _id: id,
+        history: [],
+        replication_id_version: REPLICATION_ID_VERSION
+      };
     }
     throw err;
   }).then(function (doc) {
     if (returnValue.cancelled) {
       return;
     }
+    // Filter out current entry for this replication
+    doc.history = (doc.history || []).filter(function(item) {
+      return item.session_id !== session;
+    });
+
+    // Add the latest checkpoint to history
+    doc.history.unshift({
+      last_seq: checkpoint,
+      session_id: session
+    });
+
+    // Just take the last pieces in history, to
+    // avoid really big checkpoint docs.
+    // CouchDB 2.0 has a more involved history pruning,
+    // but let's go for the simple version for now.
+    doc.history = doc.history.slice(0, 5);
```

Comment:
Maybe the `5` should better be defined as a constant like variable, eg. `CHECKPOINT_HISTORY_SIZE`?